src/Tabs/Tabs.js

```diff
@@ -0,0 +1,266 @@
+        const { stackedTabs, activeTabIndex } = this.props;
+        return {
+            stacked: stackedTabs,
+            active: index === activeTabIndex,
+            addTabRef: this.addTabRef,
+        };
+    }
+
+    renderTabItems() {
+        const { tabItems } = this.props;
+        return tabItems.map((tab, index) => (
+            <Tab key={`tab-${index}`} {...this.getAdditionalTabProps(index)} {...tab} />
+        ));
+    }
+
+    renderChildren() {
+        const { children } = this.props;
+        return Children.map(children, (child, index) =>
+            cloneElement(child, this.getAdditionalTabProps(index))
+        );
+    }
+
+    // Rendering
+    renderTabBar() {
+        const { cluster, contained, children } = this.props;
+
+        const { showTabIndicator } = this.state;
+
+        const className = bem.e(
+            'tab-container',
+            // A scrollable tabBar cannot be clustered
+            { [`cluster-${cluster}`]: cluster && contained },
+            { contained }
+        );
+
+        return (
+            <div className={className}>
+                {children ? this.renderChildren() : this.renderTabItems()}
+                <TabIndicator
+                    getActiveTabRef={this.getActiveTabRef}
+                    visible={showTabIndicator}
+                />
+            </div>
+        );
+    }
+
+    render() {
+        const { scrolledToStart, scrolledToEnd } = this.state;
+        const { position, contained } = this.props;
+        let tabBar = this.renderTabBar();
+
+        if (!contained) {
+            tabBar = (
+                <Fragment>
+                    <button
+                        onClick={this.scrollLeft}
+                        className={bem.e('scroll-button', { hidden: scrolledToStart })}
+                    >
+                        <Icon name="keyboard_arrow_left" />
+                    </button>
+                    <div className={bem.e('scroll-box')} ref={this.setScrollBoxRef}>
+                        <div className={bem.e('scroll-area')} ref={this.setScrollAreaRef}>
+                            {tabBar}
+                        </div>
+                    </div>
+                    <button
+                        onClick={this.scrollRight}
+                        className={bem.e('scroll-button', { hidden: scrolledToEnd })}
+                    >
+                        <Icon name="keyboard_arrow_right" />
+                    </button>
+                </Fragment>
+            );
+        }
+
+        return <div className={bem.b(position)}>{tabBar}</div>;
+    }
+}
+
+Tabs.propTypes = {
+    activeTabIndex: PropTypes.number.isRequired,
+    tabItems: PropTypes.array,
+    position: PropTypes.oneOf(['relative', 'fixed', 'absolute', 'sticky']),
+    contained: PropTypes.bool,
+    cluster: PropTypes.oneOf([null, 'left', 'centered', 'right']),
+    stackedTabs: PropTypes.bool,
+    children: (props, propName, componentName) => {
```

Comment:
Some custom prop-types logic here, because if the children of `Tabs` are not `Tab` components, everything would fall apart. I was surprised there isn't some type of built-in PropType that could check this. I tried `.instanceOf(Tab)` but this didn't seem to work. Any suggestions?

Reply:
`children: PropTypes.arrayOf(Tab),` should do the trick.